apis/components/v1alpha1/datasciencepipelines/types.go

```diff
@@ -0,0 +1,21 @@
+// +kubebuilder:object:generate=true
+
+// Package datasciencepipelines provides a set of types used for DataSciencePipelines component
+package datasciencepipelines
+
+import operatorv1 "github.com/openshift/api/operator/v1"
+
+type PreloadedPipelinesSpec struct {
+	InstructLab InstructLabPipelineSpec `json:"instructLab,omitempty"`
+}
+
+type InstructLabPipelineSpec struct {
+	// Set to one of the following values:
+	//
+	// - "Managed" : TODO
+	// - "Removed" : TODO
+	//
+	// +kubebuilder:validation:Enum=Managed;Removed
+	// +kubebuilder:default=Removed
+	State operatorv1.ManagementState `json:"state"`
```

Comment:
@HumairAK I kept `ManagementState` as a type for this first iteration, if you think we should use something different let me know. Also, if you have any hint about the documentation to put on this field, that would be very appreciated